cmd/chain-indexing/syncmanager.go

```diff
@@ -206,18 +214,40 @@ func (manager *SyncManager) Run() error {
 	tracker.Subscribe(blockHeightCh)
 
 	for {
-		if manager.latestBlockHeight == nil {
-			manager.logger.Info("the chain has no block yet")
-		} else {
-			if err := manager.SyncBlocks(*manager.latestBlockHeight); err != nil {
-				manager.logger.Errorf("error synchronizing blocks to latest height %d: %v", *manager.latestBlockHeight, err)
-				<-time.After(5 * time.Second)
+		isRetry := false
+		operation := func() error {
+			if manager.latestBlockHeight == nil {
+				manager.logger.Info("the chain has no block yet")
+			} else {
+				if syncErr := manager.SyncBlocks(*manager.latestBlockHeight, isRetry); syncErr != nil {
+					return fmt.Errorf(
+						"error synchronizing blocks to latest height %d: %v", *manager.latestBlockHeight, syncErr,
+					)
+				}
+			}
+
+			select {
+			case <-manager.shouldSyncCh:
+			case <-time.After(manager.pollingInterval):
 			}
+			return nil
+		}
+		notifyFn := func(opErr error, backoffDuration time.Duration) {
+			isRetry = true
+			manager.logger.Errorf(
+				"retrying in %s: %v", backoffDuration.String(), opErr,
+			)
 		}
 
-		select {
-		case <-manager.shouldSyncCh:
-		case <-time.After(manager.pollingInterval):
+		neverStopExponentialBackoff := backoff.NewExponentialBackOff()
+		neverStopExponentialBackoff.MaxElapsedTime = 0
+		neverStopExponentialBackoff.MaxInterval = 15 * time.Minute
```

Comment:
shall we put this as const on the top for easier configuration?

Reply:
Good point, done